data/scripts/dependency-resolver.py

```diff
@@ -144,7 +156,13 @@ def handle_python_check(args, conf):
 
     cmd = "%s %s" % (sys.executable, f.name)
     output, status = run_command(cmd)
-    makefile_vars["HAVE_PYTHON_%s" % dep] = "y" if status else "n"
+
+    if status:
+        makefile_vars["HAVE_PYTHON_%s" % dep] = "y"
+    else:
+        makefile_vars["HAVE_PYTHON_%s" % dep] = "n"
+        if required:
+            nof_deps[conf.get("pkgname")] = "python module"
```

Comment:
if pkgname is not defined, this will set nof_deps[None] = "python module", which doesn't make sense.

In such case it seems pkgname is mandatory, then use: nof_deps[conf["pkgname"]].

If you want to be safe against broken dependencies.json, write a schema and use our common function to load+verify the JSON against a schema.